### PR TITLE
ssl: Doc enhancement

### PR DESCRIPTION
**Pull request by IngelaAndin**

### CODE REVIEW

**Comment by IngelaAndin** [comment]
@ferd and @jwheare  hope this feel good now

**Comment by jwheare** [comment]
LGTM 👍

File: lib/ssl/doc/src/ssl.xml [hunk 1]
```diff
@@ -348,20 +348,35 @@
...
-	explicitly enabled by this option; they are supported/enabled
-	by the peer also.  Anonymous cipher suites are supported for
-	testing purposes only and are not be used when security
-	matters.</p>
+	<p>A list of cipher suites that should be supported</p>
+
+	<p>The function <seealso
+	marker="#cipher_suite-2"> ssl:cipher_suites/2 </seealso>
+	can be used to find all cipher suites that
+	are supported by default and all ciphers suites that may be configured.</p>
```
Comment (jwheare) [inline lib/ssl/doc/src/ssl.xml]: Oops missed one last one:

ciphers -> cipher

File: lib/ssl/doc/src/ssl.xml [hunk 2]
```diff
@@ -348,20 +348,35 @@
...
+	<seealso marker="#str_to_suite/1">ssl:str_to_suite/1</seealso>,
+	and <seealso marker="#suite_to_openssl_str/1">ssl:suite_to_openssl_str/1</seealso>
+	also exist to help creating customized cipher suite lists.</p>
+
+	<note><p>Note that TLS-1.3 and TLS-1.2 cipher suites are not overlapping
+	sets of cipher suites so to support both these versions cipher
+	suites from both versions need to be included. If supporting
+	TLS-1.3 versions prior to TLS-1.2 can not be supported.	</p></note>
+
+	<p>Non default cipher suites including anonymous cipher suites (PRE TLS-1.3) are supported for
```
Comment (christianjgreen) [inline lib/ssl/doc/src/ssl.xml]:
```suggestion
	<p>Non-default cipher suites including anonymous cipher suites (PRE TLS-1.3) are supported for
```

File: lib/ssl/doc/src/ssl.xml [hunk 3]
```diff
@@ -1424,10 +1450,17 @@ fun(srp, Username :: string(), UserState :: term()) ->
...
-      suites are supported by crypto. If no filter function is supplied for some
-      part the default behaviour is fun(Algorithm) -> true.</p>
+      returns false for any part of the cipher suite. If no filter function is supplied for some
+      part the default behaviour regards it as if there was a filter function that returned true.
+      
+      For examples see <seealso marker="ssl:using_ssl#Customizing-cipher-suits"> Customizing cipher suits </seealso> 
+      
+      Additionaly this function also filters the cipher suites to
+      exclude cipher suites not supported by the cryptolib used by the
+      OTP crypto application.  That is calling
```
Comment (christianjgreen) [inline lib/ssl/doc/src/ssl.xml]:
```suggestion
      OTP crypto application. That is calling
```

Comment (IngelaAndin) [reply lib/ssl/doc/src/ssl.xml]: Even the nitpick ones. It looks better without the extra white space :)

**Comment by IngelaAndin** [comment]
Thanks for all comments to improve the final result.

**Pull request merged**